MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InstrumentWidget.h

```diff
@@ -0,0 +1,230 @@
+#ifndef INSTRUMENTWIDGET_H_
+#define INSTRUMENTWIDGET_H_
+
+#include "WidgetDllOption.h"
+#include "MantidGLWidget.h"
+#include "InstrumentWidgetTypes.h"
+
+#include "MantidQtAPI/GraphOptions.h"
+#include "MantidQtAPI/WorkspaceObserver.h"
+#include "MantidAPI/AlgorithmObserver.h"
+
+class InstrumentActor;
+class OneCurvePlot;
+class CollapsiblePanel;
+class InstrumentWidgetTab;
+class XIntegrationControl;
+class SimpleWidget;
+class ProjectionSurface;
+class InstrumentWidgetRenderTab;
+
+// Qt forward declarations
+class QPushButton;
+class QDialog;
+class QSlider;
+class QTabWidget;
+class QLineEdit;
+class QLabel;
+class QCheckBox;
+class QComboBox;
+class QTextEdit;
+class QShowEvent;
+class QDragEnterEvent;
+class QDropEvent;
+class QStackedLayout;
+class QSettings;
+
+/**
+  \class  InstrumentWidget
+  \brief  This is the main window for the control of display on geometry
+  \author Srikanth Nagella
+  \date   September 2008
+  \version 1.0
+
+  This is a QT widget for the controls and display of instrument geometry. 
+  The user documentation can be found at http://www.mantidproject.org/MantidPlot:_Instrument_View
+  and needs to be updated whenever the instrument view functionality changes.
+
+ */
+class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InstrumentWidget : public QWidget, public MantidQt::API::WorkspaceObserver, public Mantid::API::AlgorithmObserver, public InstrumentWidgetTypes
+{
+  Q_OBJECT
+
+public:
+  enum SurfaceType{ FULL3D = 0, CYLINDRICAL_X, CYLINDRICAL_Y, CYLINDRICAL_Z, SPHERICAL_X, SPHERICAL_Y, SPHERICAL_Z, SIDE_BY_SIDE, RENDERMODE_SIZE };
+  enum Tab{RENDER = 0, PICK, MASK, TREE};
+
+  explicit InstrumentWidget(const QString& wsName, QWidget *parent=nullptr);
+  ~InstrumentWidget();
+  void init(bool resetGeometry = true, bool autoscaling = true, double scaleMin = 0.0, double scaleMax = 0.0, bool setDefaultView = true);
```

Comment:
This `init` looks suspect to me. Presumably this came from the previous implementation. This information should be provided at construction time via the constructor of the `InstrumentWidget`